zenoh_cpp_vendor/CMakeLists.txt

```
@@ -41,6 +41,8 @@ ament_vendor(zenoh_c_vendor
     "-DZENOHC_CARGO_FLAGS=${ZENOHC_CARGO_FLAGS}"
     "-DZENOHC_BUILD_WITH_UNSTABLE_API=TRUE"
     "-DZENOHC_CUSTOM_TARGET=${ZENOHC_CUSTOM_TARGET}"
+  PATCH_COMMAND
+    git apply ${CMAKE_CURRENT_SOURCE_DIR}/pin-rust-1.75.0.patch
```

The preferred way with `ament_vendor` is to rely on `PATCHES` arg: https://github.com/ament/ament_cmake/blob/b9a046bde84316c6863f76e7aa13564fd18352cb/ament_cmake_vendor_package/cmake/ament_vendor.cmake#L32-L36
```suggestion
  PATCHES ${CMAKE_CURRENT_SOURCE_DIR}/pin-rust-1.75.0.patch
```